internal/exec/engine.go

```diff
@@ -53,10 +53,10 @@ type Engine struct {
 
 // Run executes the stage of the given name. It returns true if the stage
 // successfully ran and false if there were any errors.
-func (e Engine) Run(stageName string) bool {
+func (e Engine) Run(stageName string) (bool, error) {
```

Comment:
The bool here is redundant and can be dropped completely. It's always true when the error is nil and false when the error is non-nil.